sweat/algorithms/metrics/core.py

```diff
@@ -62,142 +182,157 @@ def compute_zones(arg, **kwargs):
     return y
 
 
+def best_interval(arg, window, mask=None, value=0.0, **kwargs):
+    """Compute best interval of the stream
 
-def wpk(power, weight):
-    """Watts per kilo
+    Masking with replacement is controlled by keyword arguments
 
     Parameters
     ----------
-    power : list, ndarray, series
-    weight : number
+    arg: array-like
+    window : int
+        Duration of the interval in seconds
+    mask : array-like of bool, optional
+        default=None, which means no masking
+    value : number, optional
+        Value to use for replacement, default=0.0
 
     Returns
     -------
-    array-like
+    float
     """
 
-    rv = pd.Series(power, dtype=float)/ weight
-    rv = cast_array_to_original_type(rv, type(power))
+    y = rolling_mean(arg, window=window, mask=mask, value=value, **kwargs)
+
+    rv = np.max(y)
 
     return rv
 
 
-def mask_fill(arg, mask=None, value=0.0, **kwargs):
-    """Replace masked values
+def time_in_zones(arg, **kwargs):
+    """Time in zones
+
+    Calculate time [sec] spent in each zone
+
+    Parameters
+    ----------
+    arg : array-like, power or heartrate
+    kwargs : see zones
+
+    Returns
+    -------
+    array-like, the same type as arg
+    """
+    type_arg = type(arg)
+    z = pd.Series(compute_zones(arg, **kwargs))
+    tiz = z.groupby(z).count()
+    rv = cast_array_to_original_type(tiz, type_arg)
+
+    return rv
+
+
+def weighted_average_power(arg, mask=None, value=0.0, **kwargs):
```

Comment:
Minor (i.e. let's pick up later): could move to `metrics.power`

Reply:
indeed, it currently calculates both WAP and xPower - the choice is controlled by keyword 'type' with a default set to WAP. Let's see how it is being used and decide on the best strategy for moving to power.py

Reply:
Both wap and xpower are calculated from power right? Either way, can be fixed later.